packages/app/src/update/services/device-update.service.ts

```diff
@@ -116,6 +119,19 @@ export class DeviceUpdateService {
       return deviceRestartResponse
     }
 
+    return Result.success(true)
+  }
+
+  public async checkUpdate(): Promise<ResultObject<boolean>> {
+    if (this.beforeUpdateOsVersion === null) {
```

Comment:
Is the lack of clarity of this variable during one of the below methods introducing any problems?


I think it's only once checked during the application runtime.